src/server/pmix_server_listener.c

```diff
@@ -163,11 +164,15 @@ pmix_status_t pmix_start_listening(pmix_listener_t *lt)
             return PMIX_ERR_OUT_OF_RESOURCE;
         }
         /* fork off the listener thread */
-        pmix_server_globals.listen_thread_active = true;
-        if (0 > pthread_create(&engine, NULL, listen_thread, NULL)) {
-            pmix_server_globals.listen_thread_active = false;
-            return PMIX_ERROR;
-        }
+	if(!pmix_server_globals.listen_thread_active) {
+            if (0 > pthread_create(&engine, NULL, listen_thread, NULL)) {
+                pmix_server_globals.listen_thread_active = false;
+                return PMIX_ERROR;
+            }
+	    else {
+                pmix_server_globals.listen_thread_active = true;
+	    }
+	}
```

Comment:
I don't understand this code block. We just checked above (line 151) that the listen thread is not active, so why are we checking it again here?